### DISCUSSION TITLE
Cannot build Scotch on Windows 10

### DISCUSSION THREAD

**Discussion by ChessMastery** [scivision/mumps-superbuild#96] [Q&A]
I am using the following command:
**cmake -G Ninja -DCMAKE_C_COMPILER=icx -DCMAKE_CXX_COMPILER=icx -DCMAKE_MAKE_PROGRAM=C:\msys64\mingw64\bin\ninja.exe -Dprefix=../building_scotch -P scripts/build_scotch.cmake**
with scripts/run_cmake.cmake modified as
execute_process(
  COMMAND ${CMAKE_COMMAND}
  **-G "Ninja"**
  -B${bindir}
  -S${CMAKE_CURRENT_LIST_DIR}
  ${args}
  RESULT_VARIABLE ret
)

The output is:
`...
-- Build files have been written to: C:/Users/evgen/AppData/Local/Temp/5DKlhwtD4TMD/scotch-prefix/src/scotch-build
[6/8] Performing build step for 'scotch'
FAILED: scotch-prefix/src/scotch-stamp/Debug/scotch-build C:/Users/evgen/AppData/Local/Temp/5DKlhwtD4TMD/scotch-prefix/src/scotch-stamp/Debug/scotch-build
C:\WINDOWS\system32\cmd.exe /C "cd /D C:\Users\evgen\AppData\Local\Temp\5DKlhwtD4TMD\scotch-prefix\src\scotch-build && "C:\Program Files\CMake\bin\cmake.exe" --build . --config Debug && "C:\Program Files\CMake\bin\cmake.exe" -E touch C:/Users/evgen/AppData/Local/Temp/5DKlhwtD4TMD/scotch-prefix/src/scotch-stamp/Debug/scotch-build"
[1/517] Copy parser_yy.y to CMake binary directory
[2/517] Copy parser_ll.l to CMake binary directory
[3/517] Copy esmumps.h to include directory
[4/517] Generate parser_yy.c and parser_ly.h using Bison
FAILED: src/libscotch/parser_yy.c src/libscotch/parser_ly.h C:/Users/evgen/AppData/Local/Temp/5DKlhwtD4TMD/scotch-prefix/src/scotch-build/src/libscotch/parser_yy.c C:/Users/evgen/AppData/Local/Temp/5DKlhwtD4TMD/scotch-prefix/src/scotch-build/src/libscotch/parser_ly.h
C:\WINDOWS\system32\cmd.exe /C "cd /D C:\Users\evgen\AppData\Local\Temp\5DKlhwtD4TMD\scotch-prefix\src\scotch-build\src\libscotch && "C:\Program Files (x86)\GnuWin32\bin\bison.exe" -pscotchyy --defines="parser_ly.h" --output="parser_yy.c" parser_yy.y"
parser_yy.y:108.18-21: syntax error, unexpected identifier
[5/517] Building C object src\libscotch\CMakeFiles\dummysizes.dir\Debug\dummysizes.c.obj
FAILED: src/libscotch/CMakeFiles/dummysizes.dir/Debug/dummysizes.c.obj
C:\PROGRA~2\Intel\oneAPI\compiler\latest\bin\icx.exe  /nologo -DCOMMON_OS_WINDOWS -DCOMMON_PTHREAD -DCOMMON_RANDOM_FIXED_SEED -DINTSIZE32 -DSCOTCH_PATCHLEVEL_NUM=4 -DSCOTCH_PTHREAD -DSCOTCH_PTHREAD_MPI -DSCOTCH_RELEASE_NUM=0 -DSCOTCH_RENAME -DSCOTCH_VERSION_NUM=7 -Drestrict=__restrict -DCMAKE_INTDIR=\"Debug\"  /DWIN32 /D_WINDOWS /W3 /MDd /Zi /Ob0 /Od /RTC1 -QMD -QMT src\libscotch\CMakeFiles\dummysizes.dir\Debug\dummysizes.c.obj -QMF src\libscotch\CMakeFiles\dummysizes.dir\Debug\dummysizes.c.obj.d /Fosrc\libscotch\CMakeFiles\dummysizes.dir\Debug\dummysizes.c.obj /Fdsrc\libscotch\CMakeFiles\dummysizes.dir\Debug\ -c C:\Users\evgen\AppData\Local\Temp\5DKlhwtD4TMD\scotch-prefix\src\scotch\src\libscotch\dummysizes.c
In file included from C:\Users\evgen\AppData\Local\Temp\5DKlhwtD4TMD\scotch-prefix\src\scotch\src\libscotch\dummysizes.c:94:
C:\Users\evgen\AppData\Local\Temp\5DKlhwtD4TMD\scotch-prefix\src\scotch\src\libscotch\common.h(142,21): fatal error: 'pthread.h' file not found
  142 | #include            <pthread.h>
      |                     ^~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
ninja: build stopped: subcommand failed.
CMake Error at scripts/run_cmake.cmake:27 (message):
  scotch failed to build and install.
Call Stack (most recent call first):
  scripts/build_scotch.cmake:13 (include)`
  At first BISON was required, so I installed BISON and FLEX but got another errors like pthread.h not found. Which way can I disable pt-scotch (parallel version) and try to build sequential scotch only?

**Comment by scivision** [comment]
This was recently overhauled to auto-build Scotch as part of the MUMPS CMake project

```sh
cmake -Bbuild -DMUMPS_scotch=true

cmake --build build
```

**Reply by scivision** [reply]
you can also use presets like

```sh
cmake --workflow scotch
```
